cvmfs/swissknife_check.cc

```diff
@@ -332,15 +342,25 @@ bool CommandCheck::Find(const catalog::Catalog *catalog,
   typedef map< uint32_t, vector<catalog::DirectoryEntry> > HardlinkMap;
   HardlinkMap hardlinks;
   bool found_nested_marker = false;
+  char duplicate_lookup_result = 0;
 
   for (unsigned i = 0; i < entries.size(); ++i) {
+    entry_needs_check_ = 1;
```

Comment:
I guess this can be local to the for loop, so like
```suggestion
    const bool entry_needs_check = 
      !entries[i].checksum().IsNull() && duplicates_map_.contains(entries[i].checksum());
```

Perhaps we can also fold in here the check for external file from further down, and than simplify that if further down to `check_chunks_ && entry_needs_check_`.